rust/kernel/platdev.rs

```diff
@@ -8,15 +8,39 @@
 
 use crate::{
     bindings, c_types,
-    error::{Error, Result},
+    device::Device,
+    error::{from_kernel_err_ptr, Error, Result},
     from_kernel_result,
     of::OfMatchTable,
     str::CStr,
     types::PointerWrapper,
 };
+
 use alloc::boxed::Box;
 use core::{marker::PhantomPinned, pin::Pin};
 
+/// Platform device
+pub struct PlatformDevice(*mut bindings::platform_device);
+
+impl Device for PlatformDevice {
+    fn raw(&self) -> *const bindings::device {
+        unsafe { &(*self.0).dev }
+    }
+
+    fn mut_raw(&mut self) -> *mut bindings::device {
+        unsafe { &mut (*self.0).dev }
+    }
+}
+
+/// devm_platform_ioremap_resource - call devm_ioremap_resource() for a platform device
+pub fn devm_platform_ioremap_resource(
+    pdev: &mut PlatformDevice,
+    index: u32,
+) -> Result<*mut c_types::c_void> {
+    // SAFETY: FFI call.
```

Comment:
Doesn't the pointer need to be valid etc.?

Reply:
I think that a note about safety contract is required here.

Also this code is present here as an example:
https://github.com/Rust-for-Linux/linux/blob/8c64300cfff922c28ded18e781e3daed8444a268/rust/kernel/error.rs#L489-L493

The commit should also update the example there.

Reply:
Yeah, agreed.